File: BMICalculator/app/src/main/java/com/s15851/bmi_calc/StatisticsActivity.java
package com.s15851.bmi_calc;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.github.mikephil.charting.charts.BarChart;
import com.github.mikephil.charting.components.Description;
import com.github.mikephil.charting.components.XAxis;
import com.github.mikephil.charting.data.BarData;
import com.github.mikephil.charting.data.BarDataSet;
import com.github.mikephil.charting.data.BarEntry;
import com.github.mikephil.charting.formatter.IndexAxisValueFormatter;
import com.github.mikephil.charting.utils.ColorTemplate;

import java.util.ArrayList;

public class StatisticsActivity extends AppCompatActivity {

    BarChart barChart;
    ArrayList<BarEntry> barEntryArrayList;
    ArrayList<String> labelsNames;

    ArrayList<CoronavirusData> coronavirusDataArrayList = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_statistics);
        barChart = findViewById(R.id.barChart);
        // create new object of bare entries arraylist and labels arraylist
        barEntryArrayList = new ArrayList<>();
        labelsNames = new ArrayList<>();
        infectionStatistics();

        Button back = findViewById(R.id.btBack);
        back.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), BmiActivity.class);
                startActivity(intent);
            }
        });

        for (int i = 0; i < coronavirusDataArrayList.size(); i++) {
            String region = coronavirusDataArrayList.get(i).getRegion();
            int infected = coronavirusDataArrayList.get(i).getInfected();
            barEntryArrayList.add(new BarEntry(i, infected));
            labelsNames.add(region);
        }

        BarDataSet barDataSet = new BarDataSet(barEntryArrayList,"Number of coronavirus infections in voivodships as at 30.03.2020");
        barDataSet.setColor(ContextCompat.getColor(barChart.getContext(), R.color.colorPrimary));
        barDataSet.setValueTextSize(15);
        Description description = new Description();
        description.setText("voivodship");
        barChart.setDescription(description);
        BarData barData = new BarData(barDataSet);
        barChart.setData(barData);
        //set XAxis value formater
        XAxis xAxis = barChart.getXAxis();
        xAxis.setValueFormatter(new IndexAxisValueFormatter(labelsNames));
        //set position of labels(regions names)
        xAxis.setPosition(XAxis.XAxisPosition.TOP);
        xAxis.setDrawGridLines(false);
        xAxis.setDrawAxisLine(false);
        xAxis.setGranularity(1f);
        xAxis.setLabelCount(labelsNames.size());
        xAxis.setLabelRotationAngle(270);
        barChart.animateY(2000);
        barChart.invalidate();
    }

    private void infectionStatistics() {
        coronavirusDataArrayList.clear();
        coronavirusDataArrayList.add(new CoronavirusData("dolnośląskie", 258));
        coronavirusDataArrayList.add(new CoronavirusData("kujawsko-pomorskie", 72));
        coronavirusDataArrayList.add(new CoronavirusData("lubelskie", 114));
        coronavirusDataArrayList.add(new CoronavirusData("lubuskie", 36));
        coronavirusDataArrayList.add(new CoronavirusData("łódzkie", 202));
        coronavirusDataArrayList.add(new CoronavirusData("małopolskie", 174));
        coronavirusDataArrayList.add(new CoronavirusData("mazowieckie", 487));
        coronavirusDataArrayList.add(new CoronavirusData("opolskie", 54));
        coronavirusDataArrayList.add(new CoronavirusData("podkarpackie", 80));
        coronavirusDataArrayList.add(new CoronavirusData("podlaskie", 36));
        coronavirusDataArrayList.add(new CoronavirusData("pomorskie", 48));
        coronavirusDataArrayList.add(new CoronavirusData("śląskie", 209));
        coronavirusDataArrayList.add(new CoronavirusData("świętokrzyskie", 37));
        coronavirusDataArrayList.add(new CoronavirusData("warmińsko-mazurskie", 57));
        coronavirusDataArrayList.add(new CoronavirusData("wielkopolskie", 132));
        coronavirusDataArrayList.add(new CoronavirusData("zachodniopomorskie", 59));
    }
}

File: BMICalculator_Kotlin/app/src/main/java/com/s15851/bmicalculator/StatisticsActivity.kt
package com.s15851.bmicalculator

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import com.github.mikephil.charting.components.XAxis
import com.github.mikephil.charting.data.BarData
import com.github.mikephil.charting.data.BarDataSet
import com.github.mikephil.charting.data.BarEntry
import kotlinx.android.synthetic.main.activity_statistics.*


class StatisticsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_statistics)

        infectionStatistics()

        btBack.setOnClickListener {
            val intent = Intent(this, BmiActivity::class.java)
            startActivity(intent)
        }
    }
    private fun infectionStatistics() {
        val coronavirusData = ArrayList<BarEntry>()
        coronavirusData.add(BarEntry(0f, 258f))
        coronavirusData.add(BarEntry(1f, 72f))
        coronavirusData.add(BarEntry(2f, 114f))
        coronavirusData.add(BarEntry(3f, 36f))
        coronavirusData.add(BarEntry(4f, 202f))
        coronavirusData.add(BarEntry(5f, 174f))
        coronavirusData.add(BarEntry(6f, 487f))
        coronavirusData.add(BarEntry(7f, 54f))
        coronavirusData.add(BarEntry(8f, 80f))
        coronavirusData.add(BarEntry(9f, 36f))
        coronavirusData.add(BarEntry(10f, 48f))
        coronavirusData.add(BarEntry(11f, 209f))
        coronavirusData.add(BarEntry(12f, 37f))
        coronavirusData.add(BarEntry(13f, 57f))
        coronavirusData.add(BarEntry(14f, 132f))
        coronavirusData.add(BarEntry(15f, 59f))

        val barDataSet = BarDataSet(coronavirusData, "voivodeship")

        val labels = ArrayList<String>()
        labels.add("dolnośląskie")
        labels.add("kujawsko-pomorskie")
        labels.add("lubelskie")
        labels.add("lubuskie")
        labels.add("łódzkie")
        labels.add("małopolskie")
        labels.add("mazowieckie")
        labels.add("opolskie")
        labels.add("podkarpackie")
        labels.add("podlaskie")
        labels.add("pomorskie")
        labels.add("świętokrzyskie")
        labels.add("warmińsko-mazurskie")
        labels.add("wielkopolskie")
        labels.add("zachodniopomorskie")

//        val data = BarData(labels, barDataSet)
        val data = BarData(barDataSet)

        barDataSet.color = ContextCompat.getColor(this, R.color.colorPrimary)

        barChart.data = data
//        barChart.xAxis.isEnabled = false
        barChart.xAxis.position = XAxis.XAxisPosition.TOP
        barChart.xAxis.labelRotationAngle = 300f
        barChart.xAxis.labelCount = 16

        barChart.xAxis.enableGridDashedLine(5f, 5f, 0f)
        barChart.axisRight.enableGridDashedLine(5f, 5f, 0f)
        barChart.axisLeft.enableGridDashedLine(5f, 5f, 0f)
        barChart.description.isEnabled = false
        barChart.legend.isEnabled = true
        barChart.setPinchZoom(true)
        barChart.data.setDrawValues(false)

        barChart.animateY(2000)
    }
}

